File: ShadowgramUI/src/main/java/com/github/borz7zy/telegramm/ui/MainViewModel.java
package com.github.borz7zy.telegramm.ui;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.github.borz7zy.shadowgram.shadowgramui.R;
import com.github.borz7zy.telegramm.ui.model.DialogItem;

import org.drinkless.tdlib.TdApi;

import java.util.List;
import java.util.concurrent.ConcurrentHashMap;

public class MainViewModel extends ViewModel {

    private final MutableLiveData<Integer> currentTab = new MutableLiveData<>(R.id.nav_chats);

    public LiveData<Integer> getCurrentTab() {
        return currentTab;
    }

    public void setCurrentTab(int id) {
        currentTab.setValue(id);
    }

    private final MutableLiveData<Integer> topInset = new MutableLiveData<>();

    public LiveData<Integer> getTopInset() {
        return topInset;
    }

    public void setTopInset(int value) {
        topInset.setValue(value);
    }

    private final MutableLiveData<Integer> bottomInset = new MutableLiveData<>();

    public LiveData<Integer> getBottomInset() {
        return bottomInset;
    }

    public void setBottomInset(int value) {
        bottomInset.setValue(value);
    }

    private final ConcurrentHashMap<Long, DialogItem> dialogs = new ConcurrentHashMap<>();

    public ConcurrentHashMap<Long, DialogItem> getDialogs() {
        return dialogs;
    }

    private final ConcurrentHashMap<Long, TdApi.Chat> chatCache = new ConcurrentHashMap<>();

    public ConcurrentHashMap<Long, TdApi.Chat> getChatCache() {
        return chatCache;
    }

    private final MutableLiveData<List<DialogItem>> dialogList = new MutableLiveData<>();

    public LiveData<List<DialogItem>> getDialogList() {
        return dialogList;
    }

    public void postDialogList(List<DialogItem> list) {
        dialogList.setValue(list);
    }

    private final MutableLiveData<Boolean> dialogsLoading = new MutableLiveData<>(true);

    public LiveData<Boolean> getDialogsLoading() {
        return dialogsLoading;
    }

    public void setDialogsLoading(boolean loading) {
        dialogsLoading.setValue(loading);
    }

    private final MutableLiveData<Boolean> foldersAvailable = new MutableLiveData<>(false);

    public LiveData<Boolean> getFoldersAvailable() {
        return foldersAvailable;
    }

    public void setFoldersAvailable(boolean value) {
        if (!Boolean.valueOf(value).equals(foldersAvailable.getValue())) {
            foldersAvailable.setValue(value);
        }
    }
}

File: app/src/main/java/com/github/borz7zy/telegramm/ui/MainViewModel.kt
package com.github.borz7zy.telegramm.ui

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.github.borz7zy.telegramm.R
import com.github.borz7zy.telegramm.ui.model.DialogItem
import org.drinkless.tdlib.TdApi
import java.util.concurrent.ConcurrentHashMap

class MainViewModel : ViewModel() {
    private val _currentTab = MutableLiveData<Int>(R.id.nav_chats)
    val currentTab: LiveData<Int> get() = _currentTab

    fun setCurrentTab(id: Int) {
        _currentTab.value = id
    }

    private val _topInset = MutableLiveData<Int>()
    val topInset: LiveData<Int> get() = _topInset
    fun setTopInset(value: Int) { _topInset.value = value }

    private val _bottomInset = MutableLiveData<Int>()
    val bottomInset: LiveData<Int> get() = _bottomInset
    fun setBottomInset(value: Int) { _bottomInset.value = value}

    val dialogs = ConcurrentHashMap<Long, DialogItem>()

    val chatCache = ConcurrentHashMap<Long, TdApi.Chat>()

    private val _dialogList = MutableLiveData<List<DialogItem>>()
    val dialogList: LiveData<List<DialogItem>> get() = _dialogList

    fun postDialogList(list: List<DialogItem>) {
        _dialogList.value = list
    }

    private val _dialogsLoading = MutableLiveData<Boolean>(true)
    val dialogsLoading: LiveData<Boolean> get() = _dialogsLoading

    fun setDialogsLoading(loading: Boolean) {
        _dialogsLoading.value = loading
    }

    private val _foldersAvailable = MutableLiveData<Boolean>(false)
    val foldersAvailable: LiveData<Boolean> get() = _foldersAvailable
    fun setFoldersAvailable(value: Boolean) {
        if (_foldersAvailable.value != value) _foldersAvailable.value = value
    }
}